server/src/main/scala/com/cloudera/livy/server/interactive/InteractiveSession.scala

```diff
@@ -268,6 +288,51 @@ class InteractiveSession(
     }
   }
 
+  private def datanucleusJars(livyConf: LivyConf): ArrayBuffer[String] = {
+    if (sys.env.getOrElse("LIVY_INTEGRATION_TEST", "false").toBoolean) {
+      // datanucleus jars has already been in classpath in integration test
+      ArrayBuffer.empty
+    } else {
+      val jars = ArrayBuffer.empty[String]
+      val sparkHome = livyConf.sparkHome().get
+      val libdir =
+        if (new File(sparkHome, "RELEASE").isFile) {
+          new File(sparkHome, "lib")
+        } else {
+          new File(sparkHome, "lib_managed/jars")
+        }
+
+      if (libdir.isDirectory()) {
+        jars ++= libdir.listFiles().filter(_.getName.startsWith("datanucleus-"))
+          .map(_.getAbsolutePath)
+      }
+      require(jars.nonEmpty, "datanucleus jars can not be found")
```

Comment:
Hmmm... might be that we changed that behavior in CDH. But we don't have those jars (not in that location anyway), so this assert would hit every time on CDH, so I'd rather not have it.